src/io/iter_normalize.h

```diff
@@ -175,14 +179,16 @@ class ImageNormalizeIter : public IIterator<DataInst> {
     float illumination =
         rand_uniform(rnd_) * param_.max_random_illumination * 2 - param_.max_random_illumination;
 
-    if (param_.mean_r > 0.0f || param_.mean_g > 0.0f || param_.mean_b > 0.0f) {
-      // If the input has 3 channels, we substract the mean value on each
-      if (data.shape_[0] == 3) {
-        data[0] -= param_.mean_r;
+    if (param_.mean_r > 0.0f || param_.mean_g > 0.0f ||
+        param_.mean_b > 0.0f || param_.mean_a > 0.0f) {
+      // substract mean per channel
+      data[0] -= param_.mean_r;
+      if (data.shape_[0] >= 3) {
         data[1] -= param_.mean_g;
         data[2] -= param_.mean_b;
-      } else {
-        data[0] -= param_.mean_r;
```

Comment:
The codes are still buggy. If it is gray image, the mean will be subtracted twice.... @tqchen @piiswrong